test/unit/util/statusPoll.test.js

```diff
@@ -0,0 +1,204 @@
+const { describe, beforeEach, afterEach, it } = require('mocha')
+const sinon = require('sinon')
+const { expect } = require('chai')
+
+const { serviceState, startStatusHandler, buildCombinedHandler } = require('../../../app/util/statusPoll')
+
+const okStatus = (i) => ({
+  status: serviceState.UP,
+  detail: i,
+})
+
+const withFakeTimesForEvery = function () {
+  beforeEach(function () {
+    this.clock = sinon.useFakeTimers()
+  })
+  afterEach(function () {
+    this.clock.restore()
+  })
+}
+
+const stopAfterEvery = function () {
+  afterEach(function () {
+    if (this.handler) {
+      this.handler.close()
+    }
+  })
+}
+
+const pollingPeriodMs = 1000
+const serviceTimeoutMs = 100
+
+describe('startStatusHandler', function () {
+  withFakeTimesForEvery()
+  stopAfterEvery()
+
+  it('should get the service status immediately', async function () {
+    this.handler = await startStatusHandler({
```

Comment:
something new again, the usage of `this` in the test suites. Was always using `let` or `const`

Reply:
So in `mocha` the `this` constant is always consistent as long as you use `function`s rather than `() =>`. It's one thing to watch out for